opm/simulators/linalg/findOverlapRowsAndColumns.hpp

```diff
@@ -22,59 +22,94 @@
 
 #include <vector>
 #include <utility>
+#include <opm/grid/common/WellConnections.hpp>
 
 namespace Opm
 {
 namespace detail
 {
+    /// \brief Find cell IDs for wells contained in local grid.
+    ///
+    /// Cell IDs of wells stored in a graph, so it can be used to create 
+    /// an adjacency pattern. Only relevant when the UseWellContribusion option is set to true
+    /// \tparam The type of the DUNE grid.
+    /// \tparam Well vector type
+    /// \param grid The grid where we look for overlap cells.
+    /// \param wells List of wells contained in grid.
+    /// \param useWellConn Boolean that is true when UseWellContribusion is true
+    /// \param wellGraph Cell IDs of well cells stored in a graph.
+    template<class Grid, class W>
+    void setWellConnections(const Grid& grid, const W& wells, bool useWellConn, std::vector<std::set<int>>& wellGraph)
+    {
+        if ( grid.comm().size() > 1) 
+        {
+            wellGraph.resize(grid.numCells());
+
+            if (useWellConn) {
+                const auto& cpgdim = grid.logicalCartesianSize();
+
+                std::vector<int> cart(cpgdim[0]*cpgdim[1]*cpgdim[2], -1);
```

Comment:
It seems like we have been using CpGrid special function here before also, So this should not be a problem.

In the long run we should not use special functions of `CpGrid` like `ogicalCartesianSize` or `globalCell` directly. Please take a look at ebos/ecltransmissibility.hh for another way using the vanguard stuff:
```
const auto& cartMapper = vanguard_.cartesianIndexMapper();
const auto& cpgdim = cartMapper.cartesianDimensions()
```
That is for a future PR, I guess.